engine/src/ExternalAgentInterface/vb_engine_EA_interface.c

```diff
@@ -551,7 +551,8 @@ t_VB_engineErrorCode VbEngineEAInterfaceSendFrame(
 
   if (result == VB_ENGINE_ERROR_NONE)
   {
-    memcpy((char *)msg->eaPayload.msg, payload, payloadLength);
+    if (payload && payloadLength > 0)
```

Comment:
Please, add parenthesis for each check.
Send message to VB only if conditions are met:

```suggestion
    if ((payload != NULL) && (payloadLength > 0))
    {
        memcpy((char *)msg->eaPayload.msg, payload, payloadLength);
        
         // Send frame
        VbEAMsgSend(&thisDriver->vbEAConnDesc, msg);
    }
```

Reply:
I think I tried that approach (not sending anything) first and it caused regression in functionality.

Reply:
oh, maybe I missed something, ok. 
So, can you please just change the "if" to check explicitly with NULL and using () and {}?

Reply:
yes, will do.